### PR TITLE
misc cleanups

### PR DESCRIPTION
**Pull request by Diapolo**
See individual commit-msg :).

### CODE REVIEW

**Comment by Diapolo** [comment]
Ping @theuni

**Comment by Diapolo** [comment]
Ping² @theuni

**Comment by theuni** [comment]
@Diapolo weird. Last time you pinged, I got the mail, but the PR itself 404'd at github. This time it works fine.

**Comment by theuni** [comment]
@Diapolo please don't mess with the class/struct ordering. That's not something worth going back and forth about.

**Comment by Diapolo** [comment]
@theuni Removed the struct change and will not add it again :).

**Comment by Diapolo** [comment]
@theuni I updated the univalue pull and added another commit, let's see if this is better now :).

**Comment by theuni** [comment]
Looks good other than the "header group cleanups" commit, which we're not enforcing anymore. Remove that one please, then I'll pull.

**Comment by Diapolo** [comment]
I'd like to ask @laanwj, if even the "header group cleanups" are not wanted anymore. I don't try to sort them by alphabet, but it makes some sense to keep the header groups (own, std then Qt/boost/etc.)

**Comment by Diapolo** [comment]
@theuni Can you rebase your branch to current master so the diff is easier to see here :)? Still waiting for @laanwj to sneak in and comment the header group odering change.

**Comment by theuni** [comment]
@Diapolo The whole point of the trivial tree is to avoid bothering devs with these things.

Keeping includes in groups so that system includes always come after ours is fine. If you'll limit your changes to that I can pull it in.

**Comment by Diapolo** [comment]
@theuni I fully agree to your statement and this is what the "[Trivial] header group cleanups" does. It groups system includes after ours and keeps a minimal form of ordering, nothing more.

**Comment by Diapolo** [comment]
@laanwj @jonasschnelli Any consensus on the "[Trivial] header group cleanups" pull?

File: src/rpcprotocol.h [hunk 1]
```diff
@@ -10,13 +10,14 @@
...
 #include <string>
+
 #include <boost/iostreams/concepts.hpp>
 #include <boost/iostreams/stream.hpp>
 #include <boost/asio.hpp>
 #include <boost/asio/ssl.hpp>
 #include <boost/filesystem.hpp>
 
-#include "univalue/univalue.h"
+class UniValue;
```
Comment (jonasschnelli) [inline src/rpcprotocol.h:20]: What's the benefit of this forward declaration?

**Comment by jonasschnelli** [comment]
ACK for fc1d5d6c7852b997873bfd8f166410855d477f36
ACK for ddbe459d0ee345a2bfb7a7bac7007826bbadb7c0

Rest NACK. Lets just stop header includes arranging. It's an endless Sisyphean task.

File: src/rpcprotocol.h (hunk 1 above)
Comment (Diapolo) [inline src/rpcprotocol.h:20]: That was something @theuni suggested doing. As it removes the need for a header include in a header.

**Comment by Diapolo** [comment]
I don't understand the argument, as I am doing that sisyphean task with great love ;). Also I think the files are ugly to read and perhaps wildly used include orderings could perhaps even cause problems. I've seen pulls with duplicate includes quite a few times... guess who fixed them.

**Comment by theuni** [comment]
@Diapolo Every touch of those file means that all of its dependents need to be recompiled, and causes potential merge conflicts for someone else. It's great that you do it with love, but it's worth evaluating whether it's worth doing at al. "ugly to read" and "duplicate includes" are cosmetic issues that will never be completely fixed, precisely because they don't need to be.

**Pull request closed without merging**